heron/scheduler-core/src/java/com/twitter/heron/scheduler/RuntimeManagerRunner.java

```diff
@@ -177,9 +185,29 @@ void killTopologyHandler(String topologyName) throws TopologyRuntimeManagementEx
    * Handler to update a topology
    */
   @VisibleForTesting
-  void updateTopologyHandler(String topologyName, String newParallelism)
+  void updateTopologyHandler(String topologyName, Config updateConfig)
       throws TopologyRuntimeManagementException, PackingException, UpdateDryRunResponse {
     assert !potentialStaleExecutionData;
+    String newParallelism = updateConfig.getStringValue(RUNTIME_MANAGER_COMPONENT_PARALLELISM_KEY);
+    String userRuntimeConfig = updateConfig.getStringValue(RUNTIME_MANAGER_RUNTIME_CONFIG_KEY);
+
+    if (newParallelism != null && !newParallelism.isEmpty()) {
```

Comment:
@nwangtw - what happens if both the update component parallelism and also user run time config is provided simultaneously.

Reply:
This seems to be the case across the board.